atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/keyvalue/api/watch/LockWatchEventCacheIntegrationTest.java

```diff
@@ -467,18 +470,20 @@ public void newEventsStartingWithTheSameVersionAsCurrentDoesNotThrow() {
     }
 
     @Test
-    public void timestampEventsRetentionedThrows() {
+    public void eventsExceedingCacheSizeAreNotRetentionedForActiveTransactions() {
         createEventCache(1);
         setupInitialState();
-        eventCache.processStartTransactionsUpdate(ImmutableSet.of(), SUCCESS);
+        eventCache.processStartTransactionsUpdate(TIMESTAMPS_2, SUCCESS);
 
-        assertThatThrownBy(() -> eventCache.getUpdateForTransactions(TIMESTAMPS, Optional.empty()))
-                .isExactlyInstanceOf(TransactionLockWatchFailedException.class)
-                .hasMessage("Events do not enclose the required versions");
+        TransactionsLockWatchUpdate update = eventCache.getUpdateForTransactions(TIMESTAMPS_2, Optional.empty());
 
-        assertThatThrownBy(() -> eventCache.getCommitUpdate(START_TS_1))
-                .isExactlyInstanceOf(TransactionLockWatchFailedException.class)
-                .hasMessage("start or commit info not processed for start timestamp");
+        assertThat(update.clearCache()).isTrue();
```

Comment:
Let's add some `.as()` to these assertions, so poor reviewers don't have their head explode figuring out what is going on